scipy/spatial/transform/_rigid_transformation.pyx

```diff
@@ -0,0 +1,1931 @@
+    @classmethod
+    def from_translation(cls, translation):
+        """Initialize from a translation numpy array, without a rotation.
+
+        When applying this transformation to a vector ``v``, the result is the
+        same as if the translation and vector were added together. If ``t`` is
+        the displacement vector of the translation, then:
+
+        ``Tf.from_translation(t).apply(v) == t + v``
+
+        Parameters
+        ----------
+        translation : array_like, shape (N, 3) or (3,)
+            A single translation vector or a stack of translation vectors.
+
+        Returns
+        -------
+        transformation : `RigidTransformation` instance
+
+        Examples
+        --------
+        >>> from scipy.spatial.transform import RigidTransformation as Tf
+        >>> import numpy as np
+
+        Creating a transformation from a single translation vector:
+
+        >>> t = np.array([2, 3, 4])
+        >>> t + np.array([1, 0, 0])
+        array([3, 3, 4])
+        >>> tf = Tf.from_translation(t)
+        >>> tf.apply([1, 0, 0])
+        array([3., 3., 4.])
+        >>> tf.single
+        True
+
+        The top 3x1 points in the rightmost column of the transformation matrix
+        is the translation vector:
+
+        >>> tf.as_matrix()
+        array([[1., 0., 0., 2.],
+               [0., 1., 0., 3.],
+               [0., 0., 1., 4.],
+               [0., 0., 0., 1.]])
+        >>> np.allclose(tf.as_matrix()[:3, 3], t)
+        True
+
+        Creating multiple transformations from a stack of translation vectors:
+
+        >>> t = np.array([[2, 3, 4], [1, 0, 0]])
+        >>> t + np.array([1, 0, 0])
+        array([[3, 3, 4],
+               [2, 0, 0]])
+        >>> tf = Tf.from_translation(t)
+        >>> tf.apply([1, 0, 0])
+        array([[3., 3., 4.],
+               [2., 0., 0.]])
+        >>> np.allclose(tf.as_matrix()[:, :3, 3], t)
+        True
+        >>> tf.single
+        False
+        >>> len(tf)
+        2
+        """
+        translation = np.asarray(translation, dtype=float)
+
+        if (translation.ndim not in [1, 2]
+                or translation.shape[0] == 0
+                or translation.shape[-1] != 3):
+            raise ValueError("Expected `translation` to have shape (3,), or (N, 3), "
+                             f"got {translation.shape}.")
+
+        # If a single translation vector is given, convert it to a 2D 1 x 3 matrix
+        single = False
+        if translation.shape == (3,):
+            translation = translation[None, :]
+            single = True
+
+        num_translations = translation.shape[0]
+
+        matrix = np.repeat(np.eye(4, dtype=float)[None, :, :], num_translations, axis=0)
+        matrix[:, :3, 3] = translation
+
+        if single:
+            matrix = matrix[0]
+        return cls(matrix, normalize=False, copy=False)
+
+    @cython.embedsignature(True)
+    @classmethod
+    def from_components(cls, translation, rotation):
```

Comment:
Maybe a small reason to put rotation first, to have some alignment with exponential coordinates, rotation being more complex object and being the first introduced in scipy.

Not substantial arguments, but still.

Reply:
I would prefer to keep the current ordering as discussed up here: https://github.com/scipy/scipy/pull/22267#discussion_r1922911628

Reply:
Sounds convincing.